infrastructure/resource_group_level/modules/portal-ui.bicep

```diff
@@ -0,0 +1,220 @@
+param appServicePlanEgressSubnetId string
+param appServicePlanId string
+param privateEndpointsSubnetId string
+param logAnalyticsWorkspaceId string
+param externalApiHostname string
+param tfmApiHostname string
+param portalApiHostname string
+param redisName string
+
+var dockerImageName = '${containerRegistryName}.azurecr.io/portal-ui:${environment}'
+var dockerRegistryServerUsername = 'tfs${environment}'
+
+// These values are taken from GitHub secrets injected in the GHA Action
+@secure()
+param secureSettings object = {
+
+}
+
+// These values are taken from an export of Configuration on Dev (& validating with staging).
+@secure()
+param additionalSecureSettings object = {
+  DOCKER_REGISTRY_SERVER_PASSWORD: 'test-value'
+  DTFS_CENTRAL_API_KEY: 'test-value'
+  EXTERNAL_API_KEY: 'test-value'
+  PORTAL_API_KEY: 'test-value'
+  TFM_API_KEY: 'test-value'
+}
+
+// These values are taken from GitHub secrets injected in the GHA Action
+@secure()
+param secureConnectionStrings object = {
+  COMPANIES_HOUSE_API_KEY : 'test-value'
+  SESSION_SECRET: 'test-value'
+}
+
+// These values are taken from an export of Connection strings on Dev (& validating with staging).
+@secure()
+param additionalSecureConnectionStrings object = {
+}
+
+// TODO:FN-425 use more recent definition?
+resource redis 'Microsoft.Cache/redis@2022-06-01' existing = {
+  name: redisName
+}
+
+var portalApiUrl = 'https://${portalApiHostname}'
+// Note that for externalApiUrl in the CLI script, http was used, but the value in the exported config was https.
+var externalApiUrl = 'https://${externalApiHostname}'
+
+
+// https://learn.microsoft.com/en-us/azure/virtual-network/what-is-ip-address-168-63-129-16
+var azureDnsServerIp = '168.63.129.16'
+
+// These values are hardcoded in the CLI scripts, derived in the script or set from normal env variables
+var settings = {
+  // from env.
+  COMPANIES_HOUSE_API_URL: 'test-value'
+
+  // derived
+  PORTAL_API_URL: portalApiUrl
+  REDIS_HOSTNAME: redis.properties.hostName
+  REDIS_PORT: redis.properties.sslPort
+  // TODO:FN-425 check if this leaks the REDIS_KEY value?
+  REDIS_KEY: redis.listKeys().primaryKey
+  EXTERNAL_API_URL: externalApiUrl
+  HTTPS: 1
+
+  // hard coded
+  WEBSITE_DNS_SERVER: azureDnsServerIp
+  WEBSITE_VNET_ROUTE_ALL: '1'
+  PORT: '5000'
+  WEBSITES_PORT: '5000'
+}
+
+// These values are taken from an export of Configuration on Dev (& validating with staging).
+var tfmApiUrl = 'https://${tfmApiHostname}'
+
+var additionalSettings = {
+  // Note that the dev & staging didn't have AI enabled
+  // If enabling, consider using APPLICATIONINSIGHTS_CONNECTION_STRING instead
+  // APPINSIGHTS_INSTRUMENTATIONKEY: applicationInsights.properties.InstrumentationKey
+  // APPLICATIONINSIGHTS_CONNECTION_STRING: applicationInsights.properties.ConnectionString
+
+  DEAL_API_URL: 'TODO:FN-425 - needed?' // doesn't look used and has same value as PORTAL_API_URL: https://tfs-dev-portal-api.azurewebsites.net
```

Comment:
I nearly just deleted it, but thought I'd at least show it was present before doing so. I'm happy to remove it now, in in the environment variable audit card at the end.

Reply:
happy for this to be tackled in a separate PR